### PR TITLE
Hotfix/increase UI coverage

### PR DESCRIPTION
**Pull request by MaxCastelCast**
This request if to fix some minor bugs and increase overall coverage :
- Add tests for the MeetingRequestViewModel
- Repair the MeetingRequestService to make it able to change the new fcm tokens and set them in the backend
- repair the NavigationTest
- Isolate the Navigation in the Main Activity to make it easier to test and maintain
- fix the issue #54

### CODE REVIEW

File: app/src/main/java/com/github/se/icebreakrr/model/message/MeetingRequestManager.kt
Comment (Kapestian) [inline app/src/main/java/com/github/se/icebreakrr/model/message/MeetingRequestManager.kt:1]: Docs: I am not sure to understand how this works. Could you provide more documentation?

File: app/src/main/java/com/github/se/icebreakrr/MainActivity.kt [hunk 1]
```diff
@@ -90,33 +93,51 @@ class MainActivity : ComponentActivity() {
...
-              MeetingRequestViewModel.Companion.Factory(
-                  profileViewModel, functions, ourUserUid, ourName))
+              functions?.let {
+                MeetingRequestViewModel.Companion.Factory(profileViewModel, it, ourUserUid, ourName)
+              })
+  val meetingRequestViewModel = MeetingRequestManager.meetingRequestViewModel
+
+  if (functions != null) {
+    IcebreakrrNavHost(
+        profileViewModel, tagsViewModel, filterViewModel, meetingRequestViewModel, Route.AUTH)
```
Comment (Kapestian) [inline app/src/main/java/com/github/se/icebreakrr/MainActivity.kt:113]: Docs: Could you add more comments in that part as it is really hard for a "notification newbie" to understand what's done here

File: app/src/main/java/com/github/se/icebreakrr/model/message/MeetingRequestService.kt [hunk 2]
```diff
@@ -38,6 +39,8 @@ class MeetingRequestService : FirebaseMessagingService() {
    */
   override fun onNewToken(token: String) {
     super.onNewToken(token)
+    MeetingRequestManager.updateRemoteToken(token)
+    Log.d("FIRESTORE KEY", "key changed by Firestore")
```
Comment (Kapestian) [inline app/src/main/java/com/github/se/icebreakrr/model/message/MeetingRequestService.kt:43]: task: remove debugging log if not needed

File: app/src/main/java/com/github/se/icebreakrr/ui/sections/Notifications.kt [hunk 3]
```diff
@@ -45,7 +44,7 @@ fun NotificationScreen(navigationActions: NavigationActions, profileViewModel: P
         BottomNavigationMenu(
             onTabSelect = { route -> navigationActions.navigateTo(route) },
             tabList = LIST_TOP_LEVEL_DESTINATIONS,
-            selectedItem = Route.NOTIFICATIONS)
+            selectedItem = navigationActions.currentRoute())
```
Comment (Kapestian) [inline app/src/main/java/com/github/se/icebreakrr/ui/sections/Notifications.kt:47]: wrong fix: Thanks for solving my issue, but on friday's meeting the coaches told us it was better to have the current screen's icon highlighted in the navigation menu. You need to unfix this and apply the selection highlight for every screen's icon. you will also need to make the icon's background less bright

image TOO WHITE!

**Review by Kapestian — CHANGES_REQUESTED** [review]
Thanks for your PR, I know fixes are not the most exiting task but you did a great job that made our app more stable and robust. 
I will just ask you to document your PR a bit more to make the review and future modification of the code easier. Also, there are some handling of null values that look odd, maybe I am getting it wrong and if it's the case don't hesitate to tell me why.

Future improvement: Using Ai and StackOverflow, I learned that some of your code could benefit from dependencies injection (e.g the MeetingRequestManager would not be needed). As I am far from being an expert in this domain, I think we should have a talk about Hilt with the team and evaluate if it is profitable to update our project with dependencies injection.

Tell me when you are ready for another round, I'll review your changes in no time.

Keep going bud!

**Comment by MaxCastelCast** [comment]
New fix incomming : Stop  the image bliping problem when you click on the Notification button when you are already in the Notification screen

**Review by Kapestian — APPROVED** [review]
Thanks for applying the changes!

**Pull request merged**